lua/telescope/actions/init.lua

```diff
@@ -287,6 +290,11 @@ actions._close = function(prompt_bufnr, keepinsert)
   actions.close_pum(prompt_bufnr)
   if not keepinsert then
     vim.cmd [[stopinsert]]
+    await_schedule(function()
+      if a.nvim_win_is_valid(original_win_id) then
+        a.nvim_win_set_cursor(original_win_id, original_cursor)
+      end
+    end)
```

Comment:
Can this not just get placed beneath the pcall on line 303 for set_current_win? I don't thin we need to do it this way.

Just `pcall(a.nvim_win_set_cursor, original_win_id, original_cursor)` or similar

Reply:
No, tried it an desn't work. `stopinsert` for some reason only takes effect afterwards. So when you set the cursor position to where it's supposed to be while being in insert mode, you'll end up one to the left when exitting.

I tried a bunch of things to fix this 😅 it's not that I like this one all too much, it was just the only one that worked.